File: bluetooth_low_energy_android/lib/src/bluetooth_gatt_server_callback_api.dart
import 'dart:typed_data';

import 'api.g.dart';

class BluetoothGattServerCallbackApi extends BluetoothGattServerCallbackHostApi
    implements BluetoothGattServerCallbackFlutterApi {
  @override
  void onCharacteristicReadRequest(int hashCode, int deviceHashCode,
      int requestId, int offset, int characteristicHashCode) {
    // TODO: implement onCharacteristicReadRequest
  }

  @override
  void onCharacteristicWriteRequest(
      int hashCode,
      int deviceHashCode,
      int requestId,
      int characteristicHashCode,
      bool preparedWrite,
      bool responseNeeded,
      int offset,
      Uint8List value) {
    // TODO: implement onCharacteristicWriteRequest
  }

  @override
  void onConnectionStateChange(
      int hashCode, int deviceHashCode, int status, int newState) {
    // TODO: implement onConnectionStateChange
  }

  @override
  void onDescriptorReadRequest(int hashCode, int deviceHashCode, int requestId,
      int offset, int descriptorHashCode) {
    // TODO: implement onDescriptorReadRequest
  }

  @override
  void onDescriptorWriteRequest(
      int hashCode,
      int deviceHashCode,
      int requestId,
      int descriptorHashCode,
      bool preparedWrite,
      bool responseNeeded,
      int offset,
      Uint8List value) {
    // TODO: implement onDescriptorWriteRequest
  }

  @override
  void onExecuteWrite(
      int hashCode, int deviceHashCode, int requestId, bool execute) {
    // TODO: implement onExecuteWrite
  }

  @override
  void onMtuChanged(int hashCode, int deviceHashCode, int mtu) {
    // TODO: implement onMtuChanged
  }

  @override
  void onNotificationSent(int hashCode, int deviceHashCode, int status) {
    // TODO: implement onNotificationSent
  }

  @override
  void onPhyRead(
      int hashCode, int deviceHashCode, int txPhy, int rxPhy, int status) {
    // TODO: implement onPhyRead
  }

  @override
  void onPhyUpdate(
      int hashCode, int deviceHashCode, int txPhy, int rxPhy, int status) {
    // TODO: implement onPhyUpdate
  }

  @override
  void onServiceAdded(int hashCode, int status, int serviceHashCode) {
    // TODO: implement onServiceAdded
  }
}

File: bluetooth_low_energy_android/android/src/main/kotlin/dev/hebei/bluetooth_low_energy_android/BluetoothGattServerCallbackApi.kt
package dev.hebei.bluetooth_low_energy_android

import android.bluetooth.*

class BluetoothGattServerCallbackApi(registrar: BluetoothLowEnergyAndroidPigeonProxyApiRegistrar) :
    PigeonApiBluetoothGattServerCallback(registrar) {
    override fun pigeon_defaultConstructor(): BluetoothGattServerCallback {
        return object : BluetoothGattServerCallback() {
            override fun onCharacteristicReadRequest(
                device: BluetoothDevice, requestId: Int, offset: Int, characteristic: BluetoothGattCharacteristic
            ) {
                super.onCharacteristicReadRequest(device, requestId, offset, characteristic)
                this@BluetoothGattServerCallbackApi.onCharacteristicReadRequest(
                    this, device, requestId.toLong(), offset.toLong(), characteristic
                ) {}
            }

            override fun onCharacteristicWriteRequest(
                device: BluetoothDevice,
                requestId: Int,
                characteristic: BluetoothGattCharacteristic,
                preparedWrite: Boolean,
                responseNeeded: Boolean,
                offset: Int,
                value: ByteArray
            ) {
                super.onCharacteristicWriteRequest(
                    device, requestId, characteristic, preparedWrite, responseNeeded, offset, value
                )
                this@BluetoothGattServerCallbackApi.onCharacteristicWriteRequest(
                    this,
                    device,
                    requestId.toLong(),
                    characteristic,
                    preparedWrite,
                    responseNeeded,
                    offset.toLong(),
                    value
                ) {}
            }

            override fun onConnectionStateChange(device: BluetoothDevice, status: Int, newState: Int) {
                super.onConnectionStateChange(device, status, newState)
                this@BluetoothGattServerCallbackApi.onConnectionStateChange(
                    this, device, status.toLong(), newState.toLong()
                ) {}
            }

            override fun onDescriptorReadRequest(
                device: BluetoothDevice, requestId: Int, offset: Int, descriptor: BluetoothGattDescriptor
            ) {
                super.onDescriptorReadRequest(device, requestId, offset, descriptor)
                this@BluetoothGattServerCallbackApi.onDescriptorReadRequest(
                    this, device, requestId.toLong(), offset.toLong(), descriptor
                ) {}
            }

            override fun onDescriptorWriteRequest(
                device: BluetoothDevice,
                requestId: Int,
                descriptor: BluetoothGattDescriptor,
                preparedWrite: Boolean,
                responseNeeded: Boolean,
                offset: Int,
                value: ByteArray
            ) {
                super.onDescriptorWriteRequest(
                    device, requestId, descriptor, preparedWrite, responseNeeded, offset, value
                )
                this@BluetoothGattServerCallbackApi.onDescriptorWriteRequest(
                    this, device, requestId.toLong(), descriptor, preparedWrite, responseNeeded, offset.toLong(), value
                ) {}
            }

            override fun onExecuteWrite(device: BluetoothDevice, requestId: Int, execute: Boolean) {
                super.onExecuteWrite(device, requestId, execute)
                this@BluetoothGattServerCallbackApi.onExecuteWrite(this, device, requestId.toLong(), execute) {}
            }

            override fun onMtuChanged(device: BluetoothDevice, mtu: Int) {
                super.onMtuChanged(device, mtu)
                this@BluetoothGattServerCallbackApi.onMtuChanged(this, device, mtu.toLong()) {}
            }

            override fun onNotificationSent(device: BluetoothDevice, status: Int) {
                super.onNotificationSent(device, status)
                this@BluetoothGattServerCallbackApi.onNotificationSent(this, device, status.toLong()) {}
            }

            override fun onPhyRead(device: BluetoothDevice, txPhy: Int, rxPhy: Int, status: Int) {
                super.onPhyRead(device, txPhy, rxPhy, status)
                this@BluetoothGattServerCallbackApi.onPhyRead(
                    this, device, txPhy.toLong(), rxPhy.toLong(), status.toLong()
                ) {}
            }

            override fun onPhyUpdate(device: BluetoothDevice, txPhy: Int, rxPhy: Int, status: Int) {
                super.onPhyUpdate(device, txPhy, rxPhy, status)
                this@BluetoothGattServerCallbackApi.onPhyUpdate(
                    this, device, txPhy.toLong(), rxPhy.toLong(), status.toLong()
                ) {}
            }

            override fun onServiceAdded(status: Int, service: BluetoothGattService) {
                super.onServiceAdded(status, service)
                this@BluetoothGattServerCallbackApi.onServiceAdded(this, status.toLong(), service) {}
            }
        }
    }
}